checkstyle-checks/src/test/java/com/sonatype/checks/ClassStructureEmptyLineCheckTest.java

```diff
@@ -0,0 +1,101 @@
+
+  private ClassStructureEmptyLineCheck check = new ClassStructureEmptyLineCheck();
+
+  @Test
+  public void testVisitToken_GoodSourceFiles() throws Exception {
+    assertNoMessages("goodClassSource.java");
+    assertNoMessages("goodInterfaceSource.java");
+    assertNoMessages("goodButEmptyClassSource.java");
+    assertNoMessages("goodMultipleClassSource.java");
+  }
+
+  @Test
+  public void testVisitToken_trailingEmptyLines() throws Exception {
+    assertOneMessage("trailingEmptyLineClassSource.java", TRAILING_EMPTY_MESSAGE);
+    assertOneMessage("trailingEmptyLinesClassSource.java", TRAILING_EMPTY_MESSAGE);
+    assertOneMessage("trailingEmptyLineInterfaceSource.java", TRAILING_EMPTY_MESSAGE);
+    assertOneMessage("trailingEmptyLinesInterfaceSource.java", TRAILING_EMPTY_MESSAGE);
+  }
+
+  @Test
+  public void testVisitToken_precedingEmptyLines() throws Exception {
+    assertOneMessage("precedingEmptyLineClassSource.java", PRECEDING_EMPTY_MESSAGE);
+    assertOneMessage("precedingEmptyLinesClassSource.java", PRECEDING_EMPTY_MESSAGE);
+    assertOneMessage("precedingEmptyLineInterfaceSource.java", PRECEDING_EMPTY_MESSAGE);
+    assertOneMessage("precedingEmptyLinesInterfaceSource.java", PRECEDING_EMPTY_MESSAGE);
+  }
+
+  @Test
+  public void testVisitToken_trailingAndPrecedingEmptyLines() throws Exception {
+    List<String> expectedMessages = Arrays.asList(
+        PRECEDING_EMPTY_MESSAGE, TRAILING_EMPTY_MESSAGE
+    );
+
+    assertTwoMessages("bothEmptyLineClassSource.java", expectedMessages);
+    assertTwoMessages("bothEmptyLinesClassSource.java", expectedMessages);
+    assertTwoMessages("bothEmptyLineInterfaceSource.java", expectedMessages);
+    assertTwoMessages("bothEmptyLinesInterfaceSource.java", expectedMessages);
+    assertTwoMessages("bothEmptyMinimalClassSource.java", expectedMessages);
+  }
+
+  private void assertNoMessages(String fileName) throws Exception {
+    File sourceFile = new File(baseDir, fileName);
+    LocalizedMessages messages = processFile(sourceFile, check);
+    assertThat(messages, is(not(nullValue())));
+    assertThat(messages.getMessages().isEmpty(), is(true));
+  }
+
+  private void assertOneMessage(String fileName, String expectedMessage) throws Exception {
+    File sourceFile = new File(baseDir, fileName);
+    LocalizedMessages messages = processFile(sourceFile, check);
+
+    assertThat(messages, is(not(nullValue())));
+    assertThat(messages.getMessages().isEmpty(), is(false));
+    assertThat(messages.getMessages().size(), is(1));
+    assertThat(messages.getMessages().first().getMessage(), is(expectedMessage));
+  }
+
+  private void assertTwoMessages(String fileName, List<String> expectedMessages) throws Exception {
+    File sourceFile = new File(baseDir, fileName);
+    LocalizedMessages messages = processFile(sourceFile, check);
+
+    assertThat(messages, is(not(nullValue())));
+    assertThat(messages.getMessages().isEmpty(), is(false));
```

Comment:
Redundant with the next assertion

Reply:
> Redundant with the next assertion

Removed in 7a854e8205cd73c8131ffe277e62ccc54c66c999